src/App/Order/Calculator/PostNl/PostNLReceiptCodeCalculator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\App\Order\Calculator\PostNl;
+
+use MyParcelNL\Pdk\App\Order\Calculator\AbstractPdkOrderOptionCalculator;
+use MyParcelNL\Pdk\Base\Service\CountryCodes;
+use MyParcelNL\Pdk\Types\Service\TriStateService;
+use MyParcelNL\Pdk\Validation\Validator\CarrierSchema;
+use MyParcelNL\Pdk\Facade\Pdk;
+
+/**
+ * When receipt code is enabled, insurance is required.
+ */
+final class PostNLReceiptCodeCalculator extends AbstractPdkOrderOptionCalculator
+{
+    public function calculate(): void
+    {
+        $shipmentOptions = $this->order->deliveryOptions->shipmentOptions;
+
+        if ($this->order->shippingAddress->cc !== CountryCodes::CC_NL) {
+            $shipmentOptions->receiptCode = TriStateService::DISABLED;
+            return;
+        }
+
+        if (TriStateService::ENABLED === $shipmentOptions->ageCheck) {
+            $shipmentOptions->receiptCode = TriStateService::DISABLED;
+            return;
+        }
+
+        if (TriStateService::ENABLED !== $shipmentOptions->receiptCode) {
```

Comment:
Is het niet logischer (ken de business logic niet 100% zeker) om deze als eerste te hebben. M.a.w. wanneer de receiptCode niet aan staat, dan maken die andere controles op land/ageCheck ook al niet meer uit toch?